### PR TITLE
Persist Package.swift file along with resolved file

### PR DESCRIPTION
**Pull request by thedavidharris**
### Short description 📝

This also persists the generated Package.swift file alongside the Package.resolved; for various bits of open source tooling, both files are required.

### Checklist ✅

- [ ✅] The code architecture and patterns are consistent with the rest of the codebase.
- [ ✅] The changes have been tested following the [documented guidelines](https://docs.tuist.io/contributors/testing-strategy/).
- [ ] The `CHANGELOG.md` has been updated to reflect the changes. In case of a breaking change, it's been flagged as such.
- [ ✅] In case the PR introduces changes that affect users, the documentation has been updated.
- [ ✅] In case the PR introduces changes that affect how the cache artifact is generated starting from the `TuistGraph.Target`, the `Constants.cacheVersion` has been updated.

### CODE REVIEW

File: Sources/TuistDependencies/SwiftPackageManager/SwiftPackageManagerInteractor.swift [hunk 1]
```diff
@@ -186,13 +194,17 @@ public final class SwiftPackageManagerInteractor: SwiftPackageManagerInteracting
...
+    let destinationPackageSwiftPath: AbsolutePath
     let destinationPackageResolvedPath: AbsolutePath
     let destinationBuildDirectory: AbsolutePath
 
     let temporaryPackageResolvedPath: AbsolutePath
     let temporaryPackageSwiftPath: AbsolutePath
 
     init(dependenciesDirectory: AbsolutePath) {
+        destinationPackageSwiftPath = dependenciesDirectory
+            .appending(component: Constants.DependenciesDirectory.lockfilesDirectoryName)
```
Comment (fortmarek) [inline Sources/TuistDependencies/SwiftPackageManager/SwiftPackageManagerInteractor.swift:206]: I'd put this in the `SwiftPackageManager` folder as suggested by @danyf90

**Review by fortmarek — COMMENTED** [review]
Great addition @thedavidharris! I think we should also put this in the documentation somewhere.

**Review by danyf90 — CHANGES_REQUESTED** [review]
If we persist the Package.swift we should persist the Carthage as well

**Comment by thedavidharris** [comment]
> If we persist the `Package.swift` we should persist the `Cartfile` as well

Good call, added similar logic/tests for Carthage.

**Review by danyf90 — APPROVED** [review]

**Review by fortmarek — APPROVED** [review]

**Pull request merged**